### PR TITLE
python3Packages.geometric: init at 0.9.7.2

### PR DESCRIPTION
**Pull request by markuskowa**
###### Motivation for this change
Geometry optimization code for molecular structures.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by markuskowa** [comment]
@GrahamcOfBorg build python3Packages.geometric

File: pkgs/development/python-modules/geometric/default.nix [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+
+  propagatedBuildInputs = [
+    networkx
+    numpy
+    scipy
+    six
+    pytest-cov
+  ];
+
+  checkInputs = [
```
Comment (sheepforce) [inline pkgs/development/python-modules/geometric/default.nix:23]: I think the prefered version is to add the `pytestCheckHook`.

File: pkgs/development/python-modules/geometric/default.nix [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+    inherit pname version;
+    sha256 = "1p9vxvzvnb6nnqlxl2r75jrih9bbqqbsm5zjn3i311qq3b1whsbg";
+  };
+
+  propagatedBuildInputs = [
+    networkx
+    numpy
+    scipy
+    six
+    pytest-cov
```
Comment (sheepforce) [inline pkgs/development/python-modules/geometric/default.nix:20]: Is `pytest-cov` required to actually run geometric?

**Review by sheepforce — CHANGES_REQUESTED** [review]

File: pkgs/development/python-modules/geometric/default.nix (hunk 2 above)
Comment (markuskowa) [reply pkgs/development/python-modules/geometric/default.nix]: No it is not needed.

File: pkgs/development/python-modules/geometric/default.nix (hunk 1 above)
Comment (markuskowa) [reply pkgs/development/python-modules/geometric/default.nix:22]: Using `pytestCheckHook` does not seem to work at all. I just get test failures. (maybe I am using it wrong?).

**Comment by risicle** [comment]
`nixpkgs-review` happy, macos 10.15, nixos x86_64 & aarch64

File: pkgs/development/python-modules/geometric/default.nix (hunk 1 above)
Comment (sheepforce) [reply pkgs/development/python-modules/geometric/default.nix:22]: Yup, just tried it myself. `pytest` and `pytest-cov` simply don't execute any tests (`0 tests ran in 0.000s`) as it seems not to discover them, while `pytestCheckHook` runs them and fails due to missing .xyz files and interfaces.

Comment (sheepforce) [reply pkgs/development/python-modules/geometric/default.nix:22]: For example geometric tries to access `geometric/data/water5.xyz`, which does actually not exist. Comparing the tarball from PyPi https://pypi.org/project/geometric/#files and Geometric's GitHub https://github.com/leeping/geomeTRIC/releases/tag/0.9.7.2 actually shows that some files are missing.  With the GitHub sources it actually works.

```
  src = fetchFromGitHub {
    owner = "leeping";
    repo = "geomeTRIC";
    rev = version;
    hash = "sha256-QFpfY6tWqcda6AJT17YBEuwu/4DYPbIMJU1c9/gHjaA=";
  };

  checkInputs = [ pytestCheckHook ];
```

Comment (markuskowa) [reply pkgs/development/python-modules/geometric/default.nix]: Thanks for diving into it. I changed it to GitHub sources and pytestCheckHook.

**Comment by sheepforce** [comment]
LGTM! :sheep:

**Review by sheepforce — APPROVED** [review]

**Pull request merged**